compiler/noirc_evaluator/src/ssa/ir/function_inserter.rs

```diff
@@ -251,4 +257,22 @@ impl<'f> FunctionInserter<'f> {
             self.values.entry(*param).or_insert(*new_param);
         }
     }
+
+    /// Merge the internal mapping into the given mapping
+    /// The merge is guaranteed to be coherent because ambiguous cases are prevented
+    pub(crate) fn extract_mapping(&self, mapping: &mut HashMap<ValueId, ValueId>) {
+        for (k, v) in &self.values {
+            if mapping.contains_key(k) {
+                unreachable!("cannot merge key");
+            }
+            if mapping.contains_key(v) {
```

Comment:
We want to check `contains_key` for the value here?

Reply:
No, if the two mapping both map to the same value, I think that's fine.
However, if the value of one is the key of the other, then one mapping is somehow overwriting what the other mapping is doing, and I disallow this. Especially because mappings are constructed in reverse order, so a mapping overwriting the 'next one' is probably bad.